internal/storage/bucket_handle_test.go

```diff
@@ -256,3 +256,51 @@ func (t *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObjectNotExi
 	AssertEq(nil, obj)
 	AssertTrue(strings.Contains(err.Error(), "Error 412: Precondition failed"))
 }
+
+func (t *BucketHandleTest) TestListObjectMethodWithValidPrefix() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertNe(nil, obj.Objects)
+	AssertNe(nil, obj.CollapsedRuns)
+	AssertEq(nil, err)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithInValidPrefix() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "InvalidPrefix",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, obj.Objects)
+	AssertEq(nil, obj.CollapsedRuns)
+	AssertEq(nil, err)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithIncludeTrailingDelimiter() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: false,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, obj.Objects)
```

Comment:
same here. check for object.

Also what is the difference between this test and first check. can you differentiate with assert statements.

Reply:
Object is nil. there is data exist to check property